lib/http_request.js

```diff
@@ -9,26 +8,25 @@ var AwsHelper = require('aws-lambda-helper');
  * @param {Function} callback - a standard callback with error & response args
  * response is a JSON Object unless there is an error.
  */
-module.exports = function simple_http_request (options, callback) {
+
+const bl = require('bl');
+
+function request (options, callback) {
   require('https').request(options, function (res) {
-    res.setEncoding('utf8');
-    var resStr = '';
-    res.on('data', function (chunk) {
-      resStr += chunk;
-    }).on('end', function () {
-      var json = null;
-      var err = null;
-      try { // avoid fatal error if ONE of the http requests has invalid JSON
-        json = JSON.parse(resStr);
+    res.pipe(bl((err, body) => {
```

Comment:
@lennym would you mind adding a comment clarifying what [`bl`](https://www.npmjs.com/package/bl) is doing here? (*thanks!*)